syntax = "proto3";

service JuceEmbedGtk {
  rpc SetWindowId(WindowId) returns (Void) {}
  rpc GetParameter(Void) returns (ParameterValue) {}
  rpc SetParameter(ParameterValue) returns (Void) {}
}

message Void {}

message WindowId {
  uint64 id = 1;
}

message ParameterValue {
  float value = 1;
}

service AudioStreaming {
  rpc GetAudioStream(Void) returns (stream AudioBuffer) {}
}

message AudioBuffer {
    repeated float samples = 1;
}